network/p2p/network.go

```diff
@@ -105,7 +113,8 @@ func NewNetwork(
 // Network exposes networking state and supports building p2p application
 // protocols
 type Network struct {
-	Peers *Peers
+	Peers      *Peers
+	Validators *Validators
```

Comment:
This is technically an unrelated change so I can pull this out of the diff if we feel like this makes the review better but it's sort of in the vein of making the sampling more generic since this allows us to in the future configure sampling by stake if we had something like:

```
n := NewNetwork(...)
client := n.NewClient(0x0, WithPeerSampler(NewStakeWeightedPeerSampler(n.Validators)))
```

Technically we don't need to bind this onto the `Network` struct to do the above change but I feel like it is more consistent with the fact that `Peers` is bound to the `Network` struct + I think it looks cleaner to have it in one spot instead of the VM managing all of these + it lets us use validator stake sampling as a default if we want to go that route.

Reply:
Hm - `Peers` makes sense to live on the `Network` struct because it is modified by calls to `Connected`/`Disconnected`. It feels pretty odd to me to pass `Validators` around on the `Network` struct even though it is never directly used.